cpg-core/src/main/java/de/fraunhofer/aisec/cpg/passes/ControlFlowSensitiveDFGPass.kt

```diff
@@ -206,6 +207,23 @@ open class ControlFlowSensitiveDFGPass : Pass() {
                 previousWrites[currentNode.refersTo]?.lastOrNull()?.let {
                     currentNode.addPrevDFG(it)
                 }
+            } else if (currentNode is ForEachStatement) {
+                // The VariableDeclaration in the ForEachStatement doesn't have an initializer, so
+                // the "normal" case won't work. We handle this case separately here...
+
+                // This is what we write to the declaration
+                val iterable = currentNode.iterable as? Expression
+
+                // We wrote something to this variable declaration
+                writtenDecl =
+                    (currentNode.variable as? DeclarationStatement)?.singleDeclaration
+                        as? VariableDeclaration
+
+                writtenDecl?.let { wd ->
+                    iterable?.let { wd.addPrevDFG(it) }
+                    // Add the variable declaration to the list of previous write nodes in this path
+                    previousWrites[wd] = mutableListOf(wd)
+                }
             }
 
             // Check for loops: No loop statement with the same state as before and no write which
```

Comment:
While you are editing this file, the comment for `handleVariableDeclaration` is wrong (copy/paste) from function declaration.